From 5dec2fdbf87dc4090d2b64eb710774219ad60e01 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Thu, 28 Aug 2014 14:22:27 -0400
Subject: [cpufreq] powernow: use cpufreq_table_validate_and_show()

Message-id: <1409235879-1283-54-git-send-email-prarit@redhat.com>
Patchwork-id: 89389
O-Subject: [RHEL7.1 PATCH BZ 1134639 053/185] cpufreq: powernow: use cpufreq_table_validate_and_show()
Bugzilla: 1134369
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Mark Langsdorf <mlangsdo@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1134639

commit b147405aa8e568750bfa99501c7fa831edef47c9
Author: Viresh Kumar <viresh.kumar@linaro.org>
Date:   Mon Sep 16 18:56:27 2013 +0530

    cpufreq: powernow: use cpufreq_table_validate_and_show()

    Lets use cpufreq_table_validate_and_show() instead of calling
    cpufreq_frequency_table_cpuinfo() and cpufreq_frequency_table_get_attr().

    Signed-off-by: Viresh Kumar <viresh.kumar@linaro.org>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Cc: Lenny Szubowicz <lszubowi@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/cpufreq/powernow-k8.c b/drivers/cpufreq/powernow-k8.c
index 5725c00..c770b70 100644
--- a/drivers/cpufreq/powernow-k8.c
+++ b/drivers/cpufreq/powernow-k8.c
@@ -1156,7 +1156,7 @@ static int powernowk8_cpu_init(struct cpufreq_policy *pol)
  pr_debug("policy current frequency %d kHz\n", pol->cur);
 
  /* min/max the cpu is capable of */
- if (cpufreq_frequency_table_cpuinfo(pol, data->powernow_table)) {
+ if (cpufreq_table_validate_and_show(pol, data->powernow_table)) {
   printk(KERN_ERR FW_BUG PFX "invalid powernow_table\n");
   powernow_k8_cpu_exit_acpi(data);
   kfree(data->powernow_table);
@@ -1164,8 +1164,6 @@ static int powernowk8_cpu_init(struct cpufreq_policy *pol)
   return -EINVAL;
  }
 
- cpufreq_frequency_table_get_attr(data->powernow_table, pol->cpu);
-
  pr_debug("cpu_init done, current fid 0x%x, vid 0x%x\n",
    data->currfid, data->currvid);
 
-- 
1.7.1